rust/chains/tw_ton/src/signing_request/builder.rs

```diff
@@ -128,33 +135,25 @@ impl SigningRequestBuilder {
             .into_tw()
             .context("Invalid 'response_address' address")?;
 
+        let custom_payload = if input.custom_payload.is_empty() {
+            None
+        } else {
+            Some(input.custom_payload.to_string())
+        };
+
         let jetton_payload = JettonTransferRequest {
             query_id: input.query_id,
             jetton_amount: U256::from(input.jetton_amount),
             dest,
             response_address,
+            custom_payload,
             forward_ton_amount: U256::from(input.forward_amount),
         };
 
         Ok(TransferPayload::JettonTransfer(jetton_payload))
     }
 
-    fn custom_request(input: &Proto::CustomPayload) -> SigningResult<TransferPayload> {
-        let state_init = if input.state_init.is_empty() {
-            None
-        } else {
-            Some(input.state_init.to_string())
-        };
-
-        let payload = if input.payload.is_empty() {
-            None
-        } else {
-            Some(input.payload.to_string())
-        };
-
-        Ok(TransferPayload::Custom(TransferCustomRequest {
-            state_init,
-            payload,
-        }))
+    fn custom_request(input: &str) -> SigningResult<TransferPayload> {
+        Ok(TransferPayload::Custom(input.to_string()))
```

Comment:
It's a good practice to take an owned String (or any other object) if you need it to be owned.
So for example, if there is a function that has a custom request `String`, and it doesn't use it after `SigningRequestBuilder::custom_request`, it can simply move it. However, in you case, there will be a copy of the string.